neonUtilities/R/zipsByProduct.R

```diff
@@ -169,154 +169,12 @@ zipsByProduct <- function(dpID, site="all", startdate=NA, enddate=NA, package="b
-        
-    } else {
-      
-      # to get all data, select the zip files
-      all.zip <- grep(".zip", tmp.files[[i]]$data$files$name, fixed=T)
-      
-      # error message if there are no zips in the package
-      if(length(all.zip)==0) {
-        messages <- c(messages, paste("No zip files found for site", tmp.files[[i]]$data$siteCode,
-                                      "and month", tmp.files[[i]]$data$month, sep=" "))
-        next
-      }
-      
-      # if package==expanded, check that expanded package exists
-      # if it doesn't, download basic package
-      pk <- package
-      if(pk=="expanded") {
-        if(length(grep(pk, tmp.files[[i]]$data$files$name))==0) {
-          pk <- "basic"
-          messages <- c(messages, paste("No expanded package found for site ",
-                                        tmp.files[[i]]$data$siteCode, " and month ",
-                                        tmp.files[[i]]$data$month,
-                                        ". Basic package downloaded instead.",
-                                        sep=""))
-        }
-      }
-      
-      # subset to package
-      which.zip <- intersect(grep(pk, tmp.files[[i]]$data$files$name, fixed=T),
-                             grep(".zip", tmp.files[[i]]$data$files$name, fixed=T))
-      
-      # check again for no files
-      if(length(which.zip)==0) {
-        messages <- c(messages, paste("No basic package files found for site",
-                                      tmp.files[[i]]$data$siteCode,
-                                      "and month", tmp.files[[i]]$data$month, sep=" "))
-        next
-      }
-      
-      zip.urls <- rbind(zip.urls, cbind(tmp.files[[i]]$data$files$name[which.zip],
-                                        tmp.files[[i]]$data$files$url[which.zip],
-                                        tmp.files[[i]]$data$files$size[which.zip]))
-      
-    }
 
-  }
+  zip.urls <- getZipUrls(tmp.files, month.urls=month.urls, avg=avg, package=package,
+                         max.pub=max.pub, max.pub.site=max.pub.site, dpID=dpID, messages=messages) %>%
+    tidyr::drop_na()
 
-  # check for no files
-  if(is.null(nrow(zip.urls))) {
-    writeLines(paste0(messages[-1], collapse = "\n"))
-    stop(paste("No files found. This indicates either your internet connection failed, or the API is temporarily unavailable, or the data available for ", 
-               dpID, 
-               " are all hosted elsewhere. Check the data portal data.neonscience.org for outage alerts, and check the ", 
-               dpID, " data download page for external links.", sep=""))
-  }
-  
-  # get size info
-  zip.urls <- data.frame(zip.urls, row.names=NULL)
-  colnames(zip.urls) <- c("name", "URL", "size")
-  downld.size <- sum(as.numeric(as.character(zip.urls$size)), na.rm=T)/1e6
-  zip.urls$URL <- as.character(zip.urls$URL)
-  zip.urls$name <- as.character(zip.urls$name)
+  downld.size <- sum(as.numeric(zip.urls$size), na.rm=T)/1e6
```

Comment:
Can you make the same improvement here that you did in the AOP functions, using humanReadable() to translate the download size?

Reply:
Can I append those changes to this PR?